lms/views/basic_lti_launch.py

```diff
@@ -36,36 +36,41 @@ class BasicLTILaunchViews:
     def __init__(self, context, request):
         self.context = context
         self.request = request
+        self.lis_result_service = request.find_service(name="lis_result_sourcedid")
 
         # Configure the front-end mini-app to run.
         self.context.js_config.update({"mode": "basic-lti-launch"})
 
-        # Add config used by frontend to call `record_submission` API.
-        params = self.request.params
-        if (
-            # The outcome reporting params are typically only available when
-            # students (not teachers) launch an assignment.
-            params.get("lis_result_sourcedid")
-            and params.get("lis_outcome_service_url")
-            # This feature is initially restricted to Canvas.
-            and self._is_launched_by_canvas()
-        ):
-            self.context.js_config["submissionParams"] = {}
-
-            self._set_submission_param("h_username", context.h_user.username)
-            self._set_submission_param(
-                "lis_result_sourcedid", params.get("lis_result_sourcedid")
-            )
-            self._set_submission_param(
-                "lis_outcome_service_url", params.get("lis_outcome_service_url")
-            )
+        self._add_canvas_submission_params()
 
         # If the launch has been configured to focus on the annotations from
         # a particular user, translate that into Hypothesis client configuration.
         focused_user = self.request.params.get("focused_user")
         if focused_user:
             self._set_focused_user(focused_user)
 
+    def _add_canvas_submission_params(self):
```

Comment:
Glad we've got the word "Canvas" in here

Reply:
```suggestion
    def add_canvas_submission_params(self):
```

I don't think these need to be `_`'d since views classes don't really have public interfaces that need to be expressed (we don't write code that imports and calls view classes). Some of the other existing methods in this class don't have a leading `_`: `sync_lti_data_to_h()`, `store_lti_data()`. Unfortunately some of the other methods _do_ have `_`'s though so it's inconsistent.

We _should_ move helper methods like this to the bottom of the class, beneath the views, though. (Again some of the existing methods are grouped at the bottom of the class but some aren't.)